portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -23,52 +26,58 @@
 import com.google.gson.Gson;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Date;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that contains saves and gets comments from database. */
-@WebServlet("/data")
-public class DataServlet extends HttpServlet {
+/** Servlet that contains, saves and gets comments from database. 
+ *  Uses LoginServlet to determine if a user is logged in to
+ *  either show or hide comments.
+ */
+@WebServlet("/comments")
+public class CommentsServlet extends HttpServlet {
 
-    private List<String> comments = new ArrayList<>();
     private final Gson gson = new Gson();
+    private final UserService credentials = UserServiceFactory.getUserService();
+    private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Query find = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
-        DatastoreService data = DatastoreServiceFactory.getDatastoreService();
-        PreparedQuery results = data.prepare(find);
+        Query query = new Query("comment").addSort("date", SortDirection.DESCENDING);
+        PreparedQuery results = datastore.prepare(query);
+        response.setContentType("application/json;");
 
-        //Construct an array of comment history
-        List<String> history = new ArrayList<>();
-        for (Entity entry: results.asIterable()) {
-            String curr = (String) entry.getProperty("comment");
-            history.add(curr);
+        if(!credentials.isUserLoggedIn()) {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        } else {
+            //Construct an array of comment history
+            List<Comment> commentHistory = new ArrayList<>();
+            for (Entity entry: results.asIterable()) {
+                String content = (String) entry.getProperty("content");
+                Date date = (Date) entry.getProperty("date");
+                String email = (String) entry.getProperty("email");
+                commentHistory.add(new Comment(content, date, email));
+            }
+            response.getWriter().println(gson.toJson(commentHistory));
         }
-
-        response.setContentType("application/json;");
-        response.getWriter().println(gson.toJson(history));
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String text = request.getParameter("input");
-        comments.add(text);
-        long timestamp = System.currentTimeMillis();
+        String content = request.getParameter("input");
+        Date date = new Date();
+        String email = credentials.getCurrentUser().getEmail();
```

Comment:
Do you want to require login to post comments?

Reply:
Good point, forgot to update it. I'll merge this pull request as is right now and submit a new one to require login to post comments. I'm also going to look for into filters and mapping them onto the servlets within the next pull request too if things work out